### PR TITLE
feat(wildcard): implemented hostname wildcard

### PR DESCRIPTION
**Pull request by martinopresnik**

### CODE REVIEW

File: src/bproxy.c [hunk 1]
```diff
@@ -12,12 +12,12 @@ void conn_init(uv_stream_t *handle)
   conn_t *conn = malloc(sizeof(conn_t));
   conn->parser = malloc(sizeof(http_parser));
   conn->request = malloc(sizeof(http_request_t));
-  conn->proxy_handle = malloc(sizeof(conn_t));
```
Comment (jkuri) [inline src/bproxy.c:15]: is there a reason to remove memory allocation for proxy_handle?

Comment (martinopresnik) [reply src/bproxy.c:15]: In next line proxy_handle is set to NULL. That means allocation is lost and it's leaking memory.

File: src/bproxy.c [hunk 2]
```diff
@@ -257,11 +258,32 @@ proxy_ip_port find_proxy_config(char *hostname)
...
+      {
+        --i;
+        --conf_i;
+      }
+      if (conf_i < 0)
       {
-        ip_port.ip = pconf->ip;
-        ip_port.port = pconf->port;
-        return ip_port;
+        // If wildchart, requested hostname may be longer
```
Comment (jkuri) [inline src/bproxy.c]: minor: wildchart -> wildcard

File: src/bproxy.c (hunk 1 above)
Comment (jkuri) [reply src/bproxy.c:15]: okay, good.

File: src/bproxy.c [hunk 3]
```diff
@@ -257,11 +258,32 @@ proxy_ip_port find_proxy_config(char *hostname)
...
+        if (wildchart || i < 0)
```
Comment (jkuri) [inline src/bproxy.c]: here too, should be `wildcard`.

**Pull request merged**